### PR TITLE
Add scim enterprise user schema endpoint impl

### PR DESCRIPTION
**Pull request by mevan-karu**

### CODE REVIEW

File: modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/UserManager.java [hunk 1]
```diff
@@ -182,4 +182,10 @@ default void updateGroup(Group oldGroup, Group newGroup)
 
         throw new NotImplementedException();
     }
-}
+
+    default List<Attribute> getEnterpriseUserSchema() throws CharonException, NotImplementedException,
```
Comment (emswbandara) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/UserManager.java]: shall we add doc comments on this new API

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java [hunk 2]
```diff
@@ -80,34 +89,57 @@ public SCIMResponse get(String id, UserManager userManager, String attributes, S
     /**
      * Builds the SCIM schemas response using the provided user schema attributes.
      *
-     * @param userSchemaAttributes
+     * @param schemas
```
Comment (emswbandara) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java]: add param description

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java [hunk 3]
```diff
@@ -80,34 +89,57 @@ public SCIMResponse get(String id, UserManager userManager, String attributes, S
...
+        String schemaResponseBody = buildSchemasResponseBody(schemas).toString();
         Map<String, String> responseHeaders = getResponseHeaders();
         return new SCIMResponse(ResponseCodeConstants.CODE_OK, schemaResponseBody, responseHeaders);
     }
 
     /**
      * Builds the SCIM schemas config json representation using the provided user schema attributes.
      *
-     * @param userSchemaAttributeList
+     * @param schemas
```
Comment (emswbandara) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java]: add param description

File: modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/UserManager.java (hunk 1 above)
Comment (mevan-karu) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/extensions/UserManager.java]: Added

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java (hunk 2 above)
Comment (mevan-karu) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java]: added

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java (hunk 3 above)
Comment (mevan-karu) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java]: added

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java [hunk 4]
```diff
@@ -70,44 +73,80 @@ public SCIMResponse get(String id, UserManager userManager, String attributes, S
...
+            JSONObject enterpriseUserSchemaObject = buildEnterpriseUserSchema(schemas.get(ENTERPRISE_USER_SCHEMA_URI));
+            rootObject.put(enterpriseUserSchemaObject);
+        }
         return rootObject;
     }
 
+    /**
+     * Builds a JSON object containing enterprise user schema attribute information.
+     *
+     * @param enterpriseUserSchemaList
```
Comment (emswbandara) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java]: add param description

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java [hunk 5]
```diff
@@ -70,44 +73,80 @@ public SCIMResponse get(String id, UserManager userManager, String attributes, S
...
+     * @return
```
Comment (emswbandara) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/SchemaResourceManager.java]: define return object and their description

**Pull request merged**